Essentials/src/main/java/com/earth2me/essentials/commands/Commandhelpop.java

```diff
@@ -20,25 +21,21 @@ public Commandhelpop() {
     @Override
     public void run(final Server server, final User user, final String commandLabel, final String[] args) throws Exception {
         user.setDisplayNick();
-        final String message = sendMessage(server, user.getDisplayName(), args);
-        if (!user.isAuthorized("essentials.helpop.receive")) {
-            user.sendMessage(message);
-        }
+        sendMessage(user, server, user.getDisplayName(), args);
     }
 
     @Override
     public void run(final Server server, final CommandSource sender, final String commandLabel, final String[] args) throws Exception {
-        sendMessage(server, Console.DISPLAY_NAME, args);
+        sendMessage(null, server, Console.DISPLAY_NAME, args);
     }
 
-    private String sendMessage(final Server server, final String from, final String[] args) throws Exception {
+    private void sendMessage(final IUser user, final Server server, final String from, final String[] args) throws Exception {
         if (args.length < 1) {
             throw new NotEnoughArgumentsException();
         }
-        final String message = tl("helpOp", from, FormatUtil.stripFormat(getFinalArg(args, 0)));
-        ess.getLogger().log(Level.INFO, message);
-        ess.broadcastMessage("essentials.helpop.receive", message);
-        return message;
+        final String message = FormatUtil.stripFormat(getFinalArg(args, 0));
+        ess.getLogger().log(Level.INFO, tlLiteral("helpOp", from, message));
+        ess.broadcastTl(null, u -> !u.isAuthorized("essentials.helpop.receive") && u.equals(user), "helpOp", from, message);
```

Comment:
Correct me if I'm wrong but I'm not sure this predicate is doing the same thing as the old behaviour.

Previously we broadcasted to everyone with the `essentials.helpop.receive` permission, and *also* sent it to the sender if they don't have that permission. Now we're broadcasting it, but only to the sender if they don't have the permission?